compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1216,7 +1207,8 @@ private void validateListenerCompatibility(BLangSimpleVariable varNode, BType rh
         }
     }
 
-    private void analyzeVarNode(BLangSimpleVariable varNode, SymbolEnv env, AttachPoint.Point... attachPoints) {
+    private void analyzeVarNode(BLangSimpleVariable varNode, AnalyzerData data, AttachPoint.Point... attachPoints) {
+        SymbolEnv env = data.env;
```

Comment:
We need a good name here also. `env` we usually use within the method. But this is kind of a cache or a previous nodes env